Content.Server/Connection/Whitelist/Conditions/ConditionNotes.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Content.Server.Database;
+using Content.Shared.Database;
+using Robust.Shared.Network;
+
+namespace Content.Server.Connection.Whitelist.Conditions;
+
+public sealed partial class ConditionNotes : WhitelistCondition
+{
+    [DataField]
+    public bool IncludeExpired = false;
+
+    [DataField]
+    public NoteSeverity MinimumSeverity  = NoteSeverity.Minor;
+
+    /// <summary>
+    /// Range in days to check for notes.
+    /// </summary>
+    [DataField]
+    public int Range = int.MaxValue;
+
+    public override async Task<(bool, string)> Condition(NetUserData data)
+    {
+        var db = IoCManager.Resolve<IServerDbManager>();
+        var remarks = await db.GetAllAdminRemarks(data.UserId.UserId);
+        var notes = new List<AdminNote>();
+        foreach (var remark in remarks)
+        {
+            var note = await db.GetAdminNote(remark.Id);
```

Comment:
This is incredibly slow. Hell this code isn't even correct? Why aren't you just using the note information returned by `GetAllAdminRemarks`?